opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -1498,6 +1529,13 @@ namespace detail {
 
         asImpl().wellModel().updateWellState(dwells, dpMaxRel(), well_state);
 
+        if (asImpl().wellModel().wellCollection()->havingVREPGroups()) {
```

Comment:
Again, the same code as above, I can see why it is needed here though.

It should definitely be made into a small method...